File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/AlertDialog.kt
package com.kinetixui.ui

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

/**
 * KinetixAlertDialog — mirrors `packages/ui/src/components/alert-dialog.tsx`.
 * The React `AlertDialog` is nearly identical to `Dialog` (same overlay/
 * card/Header/Footer/Title/Description), differing only in: no free
 * dismiss (must pick `Action`/`Cancel`), and its own `Action`/`Cancel`
 * buttons styled as `buttonVariants({ variant: "Primary" })` /
 * `buttonVariants({ variant: "Outline" })`. Reused directly rather than
 * re-derived: this composes [KinetixDialog] with `dismissible = false`,
 * and reuses [KinetixDialogHeader]/[KinetixDialogFooter]/
 * [KinetixDialogTitle]/[KinetixDialogDescription] as-is (there's no
 * meaningful visual difference to justify separate copies) — the same
 * "reuse an existing component when the source is itself a thin variant"
 * call [KinetixPagination] made for [KinetixButton].
 */
@Composable
fun KinetixAlertDialog(
    visible: Boolean,
    onDismissRequest: () -> Unit,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    KinetixDialog(visible = visible, onDismissRequest = onDismissRequest, modifier = modifier, dismissible = false, content = content)
}

@Composable
fun KinetixAlertDialogAction(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    KinetixButton(onClick = onClick, modifier = modifier, variant = KinetixButtonVariant.Primary) {
        Text(text)
    }
}

@Composable
fun KinetixAlertDialogCancel(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    KinetixButton(onClick = onClick, modifier = modifier, variant = KinetixButtonVariant.Outline) {
        Text(text)
    }
}

File: packages/ui-swiftui/Sources/KinetixUI/AlertDialog.swift
//
// AlertDialog.swift — KinetixAlertDialog family.
//
// Mirrors packages/ui/src/components/alert-dialog.tsx: a KinetixDialog
// with no dismiss affordance (`dismissible: false`) — the user must pick
// an explicit action. Content chrome reuses the KinetixDialog* header /
// title / description / footer parts (the React AlertDialog* parts are
// near-identical). `KinetixAlertDialogAction` / `Cancel` are thin
// KinetixButton wrappers, matching the source's `buttonVariants` recipes
// (Primary / Outline).
//

import SwiftUI

public struct KinetixAlertDialog<Content: View>: View {
    @Binding private var isPresented: Bool
    private let content: Content

    public init(isPresented: Binding<Bool>, @ViewBuilder content: () -> Content) {
        self._isPresented = isPresented
        self.content = content()
    }

    public var body: some View {
        KinetixDialog(isPresented: $isPresented, dismissible: false) { content }
    }
}

public struct KinetixAlertDialogAction: View {
    private let title: String
    private let action: () -> Void

    public init(_ title: String, action: @escaping () -> Void) {
        self.title = title
        self.action = action
    }

    public var body: some View {
        KinetixButton(action: action) { Text(title) }
    }
}

public struct KinetixAlertDialogCancel: View {
    private let title: String
    private let action: () -> Void

    public init(_ title: String = "Cancel", action: @escaping () -> Void) {
        self.title = title
        self.action = action
    }

    public var body: some View {
        KinetixButton(variant: .outline, action: action) { Text(title) }
    }
}
